src/pool.c

```diff
@@ -699,6 +699,27 @@ account_hr(double *avg, const char *address)
     pthread_rwlock_unlock(&rwlock_acc);
 }
 
+void
+account_whr(double *avg, const char *address, const char *rig_id)
+{
+    account_t *account = NULL;
+    pthread_rwlock_rdlock(&rwlock_acc);
+    HASH_FIND_STR(accounts, address, account);
+    if (!account)
+        goto bail;
+    client_t *c = (client_t*)gbag_first(bag_clients);
+    while ((c = gbag_next(bag_clients, 0)) && sizeof(rig_id) != 0 )
```

Comment:
`sizeof(rig_id)` is an error here. It will always equal the size of a char* (e.g. 8 bytes). Just do something like
```
if (!*rig_id)
    return;
```
or similar (such as strlen), before you even get a lock above, if you want to make sure rig_id has been supplied and has data in it.